packages/next-swc/crates/core/src/relay.rs

```diff
@@ -0,0 +1,157 @@
+use std::path::PathBuf;
+
+use pathdiff::diff_paths;
+use serde::Deserialize;
+use swc_atoms::{js_word, JsWord};
+use swc_common::FileName::Real;
+use swc_common::{FileName, Span};
+use swc_ecmascript::ast::*;
+use swc_ecmascript::visit::{Fold, FoldWith};
+
+#[derive(Clone, Debug, Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub enum RelayLanguageConfig {
+    Typescript,
+    Flow,
+}
+
+#[derive(Clone, Debug, Deserialize)]
+pub struct Config {
+    pub artifact_directory: Option<PathBuf>,
+    pub language: RelayLanguageConfig,
+}
+
+impl Config {
+    fn file_extension(&mut self) -> &'static str {
+        match self.language {
+            RelayLanguageConfig::Typescript => "ts",
+            RelayLanguageConfig::Flow => "js",
+        }
+    }
+}
+
+struct Relay {
+    config: Config,
+    file_name: FileName,
+}
+
+fn pull_first_operation_name_from_tpl(tpl: TaggedTpl) -> Option<String> {
+    tpl.tpl
+        .quasis
+        .into_iter()
+        .filter_map(|quasis| {
+            let template_string_content = String::from(quasis.raw.value.to_string());
+            let split_content = template_string_content.split(" ").collect::<Vec<&str>>();
+
+            let operation = split_content
+                .chunks(2)
+                .filter_map(|slice| {
+                    if slice.len() == 1 {
+                        return None;
+                    }
+
+                    let word = slice[0];
+                    let next_word = slice[1];
+
+                    if word == "query" || word == "subscription" || word == "mutation" {
+                        return Some(String::from(next_word));
+                    }
+
+                    None
+                })
+                .next();
+
+            return operation;
+        })
+        .next()
+}
+
+fn build_require_expr_from_path(path: String) -> Expr {
+    Expr::Call(CallExpr {
+        span: Default::default(),
```

Comment:
Is it normal to default these like so?

Reply:
Yes, as it's generated by the compiler and does not have a source code to highlight on error.